client/src/containers/SignUp/SignUp.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom'
+import SmallContainer from '../../components/SmallContainer/SmallContainer';
+import InputField from '../../components/InputField/InputField';
+import { getError, hasError, isTouched } from '../../libs/validate';
+import handleErrors from './Schema';
+import './SignUp.css';
+
+const SignUp = () => {
+  const [ user, setUser ] = useState({ name: '', email: '', password: '' });
+  const [ touched, setTouched ]= useState({});
+  const [ error, setError ] = useState({});
+  
+  useEffect(() => handleErrors(user, setError), [user]);
+  
+  const handleChange = field => event => setUser({ ...user, [field]: event.target.value });
+
+  const handleBlur = (field) => (event) => setTouched({ ...touched, [field]: true});
+
+  const renderInputField = (type, label) => (
+    <InputField 
+      labelDirection="left"
+      type={type}
+      id={type}
+      label={label}
+      onChange={handleChange(type)}
+      onBlur={handleBlur(type)}
+      error={ error[type] ? getError(type, touched, error) : null}
+      value={user[type]}
+    />
+  )
+
+  return (
+    <SmallContainer>
+      <form className="has-text-centered">
+        <figure className="image is-128x128">
+          <img
+            src="https://dummyimage.com/100x100/000/fff&text=CoderHood"
+            alt="CoderHood"
+            className="brand-logo is-rounded"
+          />
+        </figure>
+
+        {renderInputField('name', 'Name')}
```

Comment:
Try to use object as function argument, as then we can know whats the arguments mean, before looking at the definition.